settings.php

```diff
@@ -262,7 +262,63 @@
     $help .= get_string('auth_fieldlock_expl', 'auth');
     $help .= get_string('auth_updateremote_expl', 'auth');
 
-    if (moodle_major_version() < '3.3') {
+    // User block and redirect feature setting section.
+    $settings->add(new admin_setting_heading('auth_saml2/blockredirectheading', get_string('blockredirectheading', 'auth_saml2'),
+        new lang_string('auth_saml2blockredirectdescription', 'auth_saml2')));
+
+    // Flagged login response options.
+    $flaggedloginresponseoptions = [
+        saml2_settings::OPTION_FLAGGED_LOGIN_NONE => get_string('flaggedresponsetypenone', 'auth_saml2'),
+        saml2_settings::OPTION_FLAGGED_LOGIN_MESSAGE => get_string('flaggedresponsetypemessage', 'auth_saml2'),
+        saml2_settings::OPTION_FLAGGED_LOGIN_REDIRECT => get_string('flaggedresponsetyperedirect', 'auth_saml2')
+    ];
+    $settings->add(new admin_setting_configselect(
+        'auth_saml2/flagresponsetype',
+        get_string('flagresponsetype', 'auth_saml2'),
+        get_string('flagresponsetype_help', 'auth_saml2'),
+        saml2_settings::OPTION_FLAGGED_LOGIN_NONE,
+        $flaggedloginresponseoptions));
+
+    // IdP attribute to look for indicating flagged account.
+    $settings->add(new admin_setting_configtext_with_maxlength(
+        'auth_saml2/flagattribute', 
+        get_string('flagattribute', 'auth_saml2'),
+        get_string('flagattribute_help', 'auth_saml2'),
+        get_string('flagattribute_default', 'auth_saml2'),
+        PARAM_TEXT,
+        null,
+        saml2_settings::FLAG_ATTRIBUTE_MAX_LENGTH));
+
+    // Value to look for indicating flag is active.
+    $settings->add(new admin_setting_configtext_with_maxlength(
+        'auth_saml2/flagvalue',
+        get_string('flagvalue', 'auth_saml2'),
+        get_string('flagvalue_help', 'auth_saml2'),
+        get_string('flagvalue_default', 'auth_saml2'),
+        PARAM_TEXT,
+        null,
+        saml2_settings::FLAG_ATTRIBUTE_MAX_LENGTH));
+
+    // Set the redirect fully qualified domain name of the http OR https scheme for flagged accounts.
+    $settings->add(new admin_setting_configtext(
+        'auth_saml2/flagredirecturl',
+        get_string('flagredirecturl', 'auth_saml2'),
+        get_string('flagredirecturl_help', 'auth_saml2'),
+        '',
+        // Regex pattern to ensure only http or https fully qualified domain names are utilised.
+        '/(https?\:\/\/(www\.)?[-a-zA-Z0-9\@\:\%\._\+\~\#\=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9\@\:\%_\+\.\~\#\?\&\/\=]*))?/'));
```

Comment:
Why don't we use PARAM_URL here?

Reply:
In testing, PARAM_URL did not block non-HTTP/HTTPS fully qualified domain names and in fact allowed plain strings through, this regex was used with the intention was to prevent admin using other schemes (like SSH etc.) to limit redirect options to only to web pages or applications. 
I will alter the `flagredirecturl_help` string to make this clear that only HTTP/S fully qualified domain names are accepted.

Reply:
Probably regex is an overkill here. But if you still want to use it, them please add some tests for it. To be 100% sure that it works as expected.